File: CS/DragDropColumns/MyFitColumnHeaderDropTargetFactory/MyFitColumnHeaderDropTarget.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.Xpf.Grid.Native;
using DevExpress.Xpf.Grid;
using System.Windows;

namespace DragDropColumns
{
    class MyFitColumnHeaderDropTarget : FitColumnHeaderDropTarget
    {
        public MyFitColumnHeaderDropTarget(GridColumnHeaderBase columnHeader) : base(columnHeader) { }

        protected override void MoveColumnTo(UIElement source, int dropIndex)
        {
            GridColumnHeader gch = source as GridColumnHeader;
            GridView.MoveColumnTo((ColumnBase)gch.DataContext, dropIndex + DropIndexCorrection, BaseGridColumnHeader.GetHeaderPresenterTypeFromGridColumnHeader(source), BaseGridColumnHeader.GetHeaderPresenterTypeFromLocalValue(AdornableElement));
        }
    }
}

File: VB/DragDropColumns/MyFitColumnHeaderDropTargetFactory/MyFitColumnHeaderDropTarget.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.Xpf.Grid.Native
Imports DevExpress.Xpf.Grid
Imports System.Windows

Namespace DragDropColumns
	Friend Class MyFitColumnHeaderDropTarget
		Inherits FitColumnHeaderDropTarget
		Public Sub New(ByVal columnHeader As GridColumnHeaderBase)
			MyBase.New(columnHeader)
		End Sub

		Protected Overrides Sub MoveColumnTo(ByVal source As UIElement, ByVal dropIndex As Integer)
			Dim gch As GridColumnHeader = TryCast(source, GridColumnHeader)
			GridView.MoveColumnTo(CType(gch.DataContext, ColumnBase), dropIndex + DropIndexCorrection, BaseGridColumnHeader.GetHeaderPresenterTypeFromGridColumnHeader(source), BaseGridColumnHeader.GetHeaderPresenterTypeFromLocalValue(AdornableElement))
		End Sub
	End Class
End Namespace
